File: leetcode_75/Array_String/js/1071.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */
var gcdOfStrings = function (str1, str2) {
    if (str2.length > str1.length) {
        return gcdOfStrings(str2, str1)
    }

    if (!str1.startsWith(str2)) {
        return ""
    }

    if (str2.length == 0) {
        return str1
    }

    return gcdOfStrings(str1.slice(str2.length, str1.length), str2)
};

File: leetcode_75/Array_String/python/1071.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if (len(str1) < len(str2)):
            return self.gcdOfStrings(str2, str1)
        elif (not str1.startswith(str2)):
            return ""
        elif (len(str2) == 0):
            return str1
        else:
            return self.gcdOfStrings(str1[len(str2):], str2)
